wodles/aws/aws_s3.py

```diff
@@ -3101,8 +3118,14 @@ def get_script_arguments():
                         help='URL for the VPC endpoint to use to obtain the STS token.')
     parser.add_argument('-se', '--service_endpoint', type=str, dest='service_endpoint', default=None,
                         help='URL for the VPC endpoint to use to obtain the logs.')
+    parser.add_argument('-ds', '--duration-seconds', type=int, dest='role_session_duration', default=None,
+                        help='The duration, in seconds, of the role session. Value can range from 900s to the max'
+                        ' session duration set for the role.')
+    parsed_args = parser.parse_args()
 
-    return parser.parse_args()
+    if parsed_args.role_session_duration is not None and parsed_args.iam_role_arn is None:
```

Comment:
Create a new `arg_valid_iam_role_duration` function and handle this there instead